camel-core/src/main/java/org/apache/camel/util/MessageHelper.java

```diff
@@ -546,7 +546,10 @@ public static String doDumpMessageHistoryStacktrace(Exchange exchange, ExchangeF
         sb.append("\n");
         sb.append("Message History\n");
         sb.append("---------------------------------------------------------------------------------------------------------------------------------------\n");
-        sb.append(String.format(MESSAGE_HISTORY_HEADER, "RouteId", "ProcessorId", "Processor", "Elapsed (ms)"));
+        String goMessageHistoryHeaeder = exchange.getContext().getGlobalOption(Exchange.MESSAGE_HISTORY_HEADER);
+        sb.append(String.format(
+                         goMessageHistoryHeaeder == null ? MESSAGE_HISTORY_HEADER : goMessageHistoryHeaeder,
+                         "RouteId", "ProcessorId", "Processor", "Elapsed (ms)"));
```

Comment:
yes, it looks cleaner. doesn't it drop out the change of this fix would go into 2.17.x releases?